test/integration/controllermanager/controllerregistration/controllerregistration_test.go

```diff
@@ -433,4 +439,194 @@ var _ = Describe("ControllerRegistration controller test", func() {
+			var (
+				policy                 = gardencorev1beta1.ControllerDeploymentPolicyAlwaysExceptNoShoots
+				controllerRegistration *gardencorev1beta1.ControllerRegistration
+			)
+
+			BeforeEach(func() {
+				controllerRegistration = &gardencorev1beta1.ControllerRegistration{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "ctrlreg-" + testID + "-",
+						Labels:       map[string]string{testID: testRunID},
+					},
+					Spec: gardencorev1beta1.ControllerRegistrationSpec{
+						Deployment: &gardencorev1beta1.ControllerRegistrationDeployment{
+							Policy: &policy,
+						},
+					},
+				}
+
+				By("Create ControllerRegistration for Shoot")
+				Expect(testClient.Create(ctx, shootControllerRegistration)).To(Succeed())
+
+				DeferCleanup(func() {
+					By("Delete ControllerRegistration for Shoot")
+					Expect(testClient.Delete(ctx, shootControllerRegistration)).To(Succeed())
+
+					By("Wait until manager has observed ControllerRegistration for Shoot deletion")
+					Eventually(func() error {
+						return mgrClient.Get(ctx, client.ObjectKeyFromObject(shootControllerRegistration), shootControllerRegistration)
+					}).Should(BeNotFoundError())
+				})
+			})
+
+			It("should delete ControllerInstallations when ControllerRegistration is deleted (even if Shoots still exist)", func() {
+				By("Create ControllerRegistration")
+				Expect(testClient.Create(ctx, controllerRegistration)).To(Succeed())
+				log.Info("Created ControllerRegistration for test", "controllerRegistration", client.ObjectKeyFromObject(controllerRegistration))
+
+				DeferCleanup(func() {
+					By("Delete ControllerRegistration")
+					Expect(testClient.Delete(ctx, controllerRegistration)).To(Succeed())
+
+					By("Wait until manager has observed ControllerRegistration deletion")
+					Eventually(func() error {
+						return mgrClient.Get(ctx, client.ObjectKeyFromObject(controllerRegistration), controllerRegistration)
+					}).Should(BeNotFoundError())
+
+					By("Delete Shoot")
+					Expect(testClient.Delete(ctx, shoot)).To(Succeed())
+				})
+
+				By("Expect finalizer to be added to ControllerRegistration")
+				Eventually(func(g Gomega) []string {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(controllerRegistration), controllerRegistration)).To(Succeed())
+					return controllerRegistration.Finalizers
+				}).Should(ConsistOf("core.gardener.cloud/controllerregistration"))
+
+				By("Expect no ControllerInstallation be created because no Shoot exists")
+				Consistently(func(g Gomega) {
+					controllerInstallationList := &gardencorev1beta1.ControllerInstallationList{}
+					g.Expect(testClient.List(ctx, controllerInstallationList, client.MatchingFields{
+						core.RegistrationRefName: controllerRegistration.Name,
+						core.SeedRefName:         seed.Name,
+					})).To(Succeed())
+					g.Expect(controllerInstallationList.Items).To(BeEmpty())
+				}).Should(Succeed())
+
+				By("Create object")
```

Comment:
```suggestion
				By("Create Shoot")
```
Just to be consistent.